tests/bluetooth/bsim_bt/bsim_test_mesh/src/test_heartbeat.c

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include "mesh_test.h"
+#include "argparse.h"
+#include "mesh/net.h"
+#include "mesh/heartbeat.h"
+
+#define LOG_MODULE_NAME test_heartbeat
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(LOG_MODULE_NAME);
+
+#define WAIT_TIME 60 /*seconds*/
+#define SUBSCRIBER_ADDR 0x00fe
+#define SUBSCRIBE_PERIOD_SEC 30
+#define PUBLISHER_ADDR_START 0x0001
+#define PUBLISH_PERIOD_SEC 1
+#define PUBLISH_MSG_CNT 10
+
+static uint16_t pub_addr = BT_MESH_ADDR_UNASSIGNED;
+
+static const struct bt_mesh_test_cfg subscribe_cfg = {
+	.addr = SUBSCRIBER_ADDR,
+	.dev_key = { 0xff },
+};
+static struct bt_mesh_test_cfg pub_cfg;
+
+static void test_publish_init(void)
+{
+	pub_cfg.addr = PUBLISHER_ADDR_START + get_device_nbr();
+	pub_cfg.dev_key[0] = get_device_nbr();
+	bt_mesh_test_cfg_set(&pub_cfg, WAIT_TIME);
+}
+
+static void test_subscribe_init(void)
+{
+	bt_mesh_test_cfg_set(&subscribe_cfg, WAIT_TIME);
+}
+
+static struct sub_context {
+	uint8_t count;
+	uint8_t min_hops;
+	uint8_t max_hops;
+} sub_ctx = {
+	.count = 0,
+	.min_hops = 0xFF,
+	.max_hops = 0,
+};
+
+static void sub_hb_recv_cb(const struct bt_mesh_hb_sub *sub, uint8_t hops, uint16_t feat)
```

Comment:
Shouldn't be `hops` and `feat` checked either?

Reply:
For `hops` I have added a check to see if it equals to one. This should correspond to the publish TTL value of zero.

For `feat` i have added a check to see if it is within the permitted feature range. Do you think this would suffice? A more exact approach will have dependencies on the prj.cfg, so I'm not quite sure how complex such a soultion would be.